eth/executionclient/multi_client.go

```diff
@@ -63,66 +74,92 @@ func NewMulti(ctx context.Context, nodeAddrs []string, contractAddr ethcommon.Ad
 		opt(multiClient)
 	}
 
-	// The underlying client may call Fatal on unsuccessful reconnection attempt.
-	// Therefore, we need to override Fatal's behavior to avoid crashing.
-	logger := multiClient.logger.WithOptions(zap.WithFatalHook(zapcore.WriteThenNoop))
-
-	for _, nodeAddr := range nodeAddrs {
-		singleClient, err := New(
-			ctx,
-			nodeAddr,
-			contractAddr,
-			WithLogger(logger),
-			WithFollowDistance(multiClient.followDistance),
-			WithConnectionTimeout(multiClient.connectionTimeout),
-			WithReconnectionInitialInterval(multiClient.reconnectionInitialInterval),
-			WithReconnectionMaxInterval(multiClient.reconnectionMaxInterval),
-			WithHealthInvalidationInterval(multiClient.healthInvalidationInterval),
-			WithSyncDistanceTolerance(multiClient.syncDistanceTolerance),
-		)
-		if err != nil {
-			return nil, fmt.Errorf("create single client: %w", err)
+	var multiErr error
+	for clientIndex := range nodeAddrs {
+		if err := multiClient.connect(ctx, clientIndex); err != nil {
+			multiClient.logger.Error("failed to connect to node",
+				zap.String("address", nodeAddrs[clientIndex]),
+				zap.Error(err))
+
+			multiErr = errors.Join(multiErr, err)
+			continue
 		}
 
-		multiClient.clients = append(multiClient.clients, singleClient)
+		multiClient.connectedCount.Add(1)
```

Comment:
I don't undestand why it needs to be atomic, never used anywhere else, only in close but only store.

Reply:
changed by me